docs/Hosting.md

```diff
@@ -0,0 +1,417 @@
+
+Implementation Sketch:
+1. Add mode detector early in startup; when active, replace tool registry with curated static tool set.
+2. Inject a response decorator adding demo warning metadata.
+3. Block any attempt to pass subscription/resource parameters (validation layer).
+4. Enforce rate limiting middleware (in-process token bucket) scoped to remote transport only.
+5. Auto-disable if an auth credential becomes available (optional; could prompt to restart in full mode).
+
+Risk Considerations:
+* Users may believe they are interacting with real Azure resources. → Mitigation: explicit labeling, banner, distinct tool names (e.g., `demo-storage-example-list`).
+* Attackers could scrape static data for amplification. → Mitigation: minimal static set, rate limiting.
+* Accidental production enablement. → Mitigation: require BOTH flag and env var (defense in depth) OR build-time compile symbol for distribution images.
+
+### 6.8 Ideal Entra ID (Authorization Code + PKCE + OBO) Flow & Gaps
+Target future architecture for per-user delegated access:
+
+**Target Sequence**
+1. MCP server advertises (in its Protocol Registration / PRM document) that it uses Microsoft Entra ID for auth (includes issuer / audience metadata pointers or discovery hints).
+2. Client connects to MCP server and retrieves the PRM document.
+3. Client discovers Entra as the Authorization Server (AS) from PRM metadata.
+4. Client constructs Entra OpenID Connect / OAuth2 metadata URLs (well-known discovery) and fetches endpoints.
+5. Client registers with the Authorization Server (dynamic client registration) OR already possesses a suitable client ID.
+6. Client initiates Authorization Code Flow with PKCE directly against Entra (user authenticates + consents).
+7. Entra issues an access token whose `aud` (audience) is the MCP server (plus optional ID token for client state).
+8. Client sends that access token to the MCP server on each request (e.g., Bearer token header / agreed channel field).
+9. MCP server validates token (issuer, signature, expiry, audience, scopes / roles, nonce / c_hash as needed).
+10. MCP server performs OAuth 2.0 On-Behalf-Of (OBO) flow to exchange the user token for downstream resource tokens (ARM, Storage, etc.).
+11. MCP server maps/associates downstream tokens to the inbound user principal (cache keyed by user object ID + scopes) and executes tool operations.
+12. Responses returned; auditing & telemetry capture per-user context.
+
+**Current Blocking Gap**: No dynamic client registration for arbitrary third-party native MCP clients.
+
+**Interim Alternatives**
+| Approach | Description | Pros | Cons / Risks |
+|----------|-------------|------|--------------|
+| Pre-registered multi-tenant Public Client | Publish a client ID (no secret) that all MCP clients can use for auth code + PKCE | Simple for ecosystem, no secret distribution | Hard to apply client-specific policies, revocation affects all, potential misuse rate limits |
```

Comment:
I don't think you can get away with this, unfortunately. What you're saying here is that you will maintain an app registration for every single MCP client that are not Microsoft, and the security team will shut that down fairly quickly. For this to work, you'd need to set up a client with multiple redirect URIs, etc. that are completely out of your control.

Reply:
Okay, will remove this as an option.